docs/example_navigation_ros_codegen.md

```diff
@@ -0,0 +1,527 @@
+---
+id: example_navigation_ros_codegen
+title: Navigation of ground vehicle using ROS codegen
+description: Nonlinear model predictive control with OpEn using ROS
+---
+
+<script type="text/x-mathjax-config">MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}});</script>
+<script type="text/javascript" async src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
+
+Example contributed by Guido Sanchez ([Github](https://github.com/gmsanchez/))
+
+## Auto-generated ground vehicle navigation for ROS
+
+Consider the navigation problem for an autonomous ground vehicle that can be modeled using the differential drive equations.
+
+### Code generation
+
+To generate a ROS package you can use opengen - OpEn's Python interface (with opengen version `0.5.0` or newer). You will have to provide certain configuration parameters, such as the package name, the node name and the rate of your node in Hz.
+
+### Install opengen 0.5.0 in a virtual environment
+
+```bash
+cd ~
+mkdir open_ros_codegen
+cd ~/open_ros_codegen
+virtualenv -p python3.6 venvopen
+source venvopen/bin/activate
+pip install opengen==0.5.0a1
+pip install matplotlib
+```
+
+### Run the example code generation program
+
+Let's make a folder for the code generation script
+
+```bash
+cd ~/open_ros_codegen
+mkdir nmpc_open
+cd nmpc_open
+```
+
+Create a file named `create_open_solver.py` inside the `nmpc_open` folder with the following content
+
```

Comment:
Perhaps it would be good to write a few sentences to state the MPC problem you are solving in this example. You could also add some comments in the Python code to explain what is happening. Although it is quite clear to us, a complete beginner might get lost (or daunted).

Reply:
You may find an example of an MPC problem statement in `example_navigation_py.md`